include: zephyr: eSPI: Add API clarifications

albertofloyd opened the pull request:
Add eSPI clarifications about eSPI bus event and OOB transactions

Fix https://github.com/zephyrproject-rtos/zephyr/issues/51095

scottwcpg commented on include/zephyr/drivers/espi.h:
```diff
@@ -450,11 +462,18 @@ typedef int (*espi_api_flash_write)(const struct device *dev,
 				    struct espi_flash_packet *pckt);
+
+/* eSPI interrupt control */
+typedef int (*espi_api_interrupt_conf)(const struct device *dev,
```
Will the interrupt configure parameter be defined by vendor drivers or defined common to all drivers.  The HW may fire many different interrupts.

albertofloyd replied:
@scottwcpg  Plan is to have some generic categories (eSPI bus events, VWs, ACPI, Port80, OOB, flash, KBC), but allow vendor specific bitmask (like pinctrl flags) so each can provide as much granularity as they see fit

albertofloyd replied:
Since new eSPI interrupt  API may take more discussion move into separate PR

AdiIntel commented on include/zephyr/drivers/espi.h:
```diff
@@ -112,8 +112,16 @@ enum espi_channel {
+
 	ESPI_BUS_EVENT_CHANNEL_READY        = BIT(1),
+	/** Indicates the eSPI HW has received a virtual wire message from eSPI host.
+	 * eSPI drivers should conveyed the eSPI virtual wire latest status.
```
should convey ..

AdiIntel commented on include/zephyr/drivers/espi.h:
```diff
@@ -112,8 +112,16 @@ enum espi_channel {
  */
 enum espi_bus_event {
+	/** Indicates the eSPI bus was reset.
+	 * eSPI drivers should conveyed the eSPI reset status in an event following eSPI reset pin convention.
```
should convey

albertofloyd replied:
corrected

AdiIntel commented on include/zephyr/drivers/espi.h:
```diff
@@ -112,8 +112,16 @@ enum espi_channel {
 enum espi_bus_event {
+	/** Indicates the eSPI bus was reset.
+	 * eSPI drivers should conveyed the eSPI reset status in an event following eSPI reset pin convention.
+	 * 0-eSPI bus in reset, 1-eSPI bus out-of-reset
```
Good clarification.

AdiIntel commented on include/zephyr/drivers/espi.h:
```diff
@@ -357,6 +365,10 @@ struct espi_request_packet {
  */
 struct espi_oob_packet {
 	uint8_t *buf;
+	/** For Tx packet, eSPI driver client shall specify length including total bytes
+	  * for eSPI OOB header and OOB payload.
```
Consider additional clarification here, whether eSPI driver should exchange data from offset byte#3 in below OOB packet format or from byte#0. 

![image](https://github.com/zephyrproject-rtos/zephyr/assets/53013931/cc1b47ef-ff28-41e7-b5a2-7e41166196a7)

IMO, Byte#0,1,2 (eSPI cycle type, {tag, Len11:8}, Len 7:0) - aren't necessary.

albertofloyd replied:
Added clarification indicating eSPI driver client and eSPI driver only exchange payload without OOB header

albertofloyd committed:
include: zephyr: drivers: eSPI API clarifications

Clarify eSPI reset bus event convention, which follows eSPI
physical reset pin.
Clarify eSPI OOB packet length expectations for eSPI driver
clients and eSPI drivers implementation

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>
